inst/stan/functions/obs_lpmf.stan

```diff
@@ -17,18 +17,27 @@ real obs_lpmf(int[] dummy, int start, int end, int[] obs, int[] sl, int[] csl,
     t = st[i];
     l = sl[i];
     vector[l] rdlh;
+    vector[l] ref_lh_i;
+    profile("model_likelihood_allocations") {
     // Find final observed/imputed expected observation
     tar_obs = imp_obs[g][t];
+    // allocate reference day effects
+    ref_lh_i = ref_lh[1:l, dpmfs[i]];
     // allocate report day effects
     rdlh = srdlh[rdlurd[t:(t + l - 1), g]];
+    }
     // combine expected final obs and date effects to get expected obs
+    profile("model_likelihood_expected_obs") {
     esnap += l;
     exp_obs[ssnap:esnap] = expected_obs(
-      tar_obs, ref_lh[1:l, dpmfs[i]], rdlh, ref_p
+      tar_obs, ref_lh_i, rdlh, ref_p
```

Comment:
The reference day effects allocation was moved out of the `expected_obs` call to be able to profile it independently.